src/self-managed-debug/src/main.rs

```diff
@@ -176,18 +188,36 @@ async fn run(context: Context) -> Result<(), anyhow::Error> {
         ),
     ]);
 
-    match create_k8s_client(context.args.k8s_context.clone()).await {
-        Ok(client) => {
-            for namespace in context.args.k8s_namespaces.clone() {
-                k8s_resource_dumper::dump_namespaced_resources(&context, &client, namespace).await;
-            }
-            k8s_resource_dumper::dump_cluster_resources(&context, &client).await;
-        }
+    let client = match create_k8s_client(context.args.k8s_context.clone()).await {
+        Ok(client) => client,
         Err(e) => {
             eprintln!("Failed to create k8s client: {}", e);
+            return Err(e);
         }
     };
 
+    for namespace in context.args.k8s_namespaces.clone() {
+        k8s_resource_dumper::dump_namespaced_resources(&context, &client, namespace).await;
+    }
+    k8s_resource_dumper::dump_cluster_resources(&context, &client).await;
+
+    if !context.args.skip_port_forward {
+        // Find the namespace, service name, local port, and target port.
+        let port_forwarding_info =
+            system_catalog_dumper::get_sql_port_forwarding_info(&client, &context.args).await?;
+
+        let _port_forward_handle = system_catalog_dumper::spawn_sql_port_forwarding_process(
+            &port_forwarding_info,
+            context.args.k8s_context.clone(),
+        );
+        // There may be a delay between when the port forwarding process starts and when it's ready to use.
+        // We wait a few seconds to ensure that port forwarding is ready.
+        tokio::time::sleep(std::time::Duration::from_secs(2)).await;
```

Comment:
The `wait` here is kind of sus, but it's what we do in mzcompose too. We could grep for the messages, but that couples us to the current kubernetes version.

Reply:
Thought about how we could implement this, and I think there needs to be two functions.

The first one like `init_port_forward` where it's an async function that waits until 
```
Forwarding from 127.0.0.1:58203 -> 8080
Forwarding from [::1]:58203 -> 8080
```
gets printed, then returns the tokio process. Then the second one like `retry_loop(tokio_process)` that repeatedly checks if the process has errored/exited – via `retry_async`– then restarts the process. Key thing being we don't await on `retry_loop`.

I don't think we can do this all in one function since if we awaited it, then that implies that the retry loop finished. Is there maybe a better implementation tho?